python/lsst/obs/base/formatters/filter.py

```diff
@@ -0,0 +1,151 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("FilterFormatter",)
+
+import yaml
+
+from typing import (
+    Any,
+    Optional,
+    Type,
+)
+
+from lsst.daf.butler.formatters.file import FileFormatter
+
+
+class FilterFormatter(FileFormatter):
+    """Read and write `~lsst.afw.image.Filter` filter information."""
+
+    extension = ".yaml"
+
+    unsupportedParameters = None
+    """This formatter does not support any parameters."""
+
+    def _readFile(self, path: str, pytype: Type[Any] = None) -> Any:
+        """Read a file from the path in YAML format.
+
+        Parameters
+        ----------
+        path : `str`
+            Path to use to open YAML format file.
+        pytype : `class`, optional
+            Not used by this implementation.
+
+        Returns
+        -------
+        data : `object`
+            Either data as Python object read from YAML file, or None
+            if the file could not be opened.
+
+        Notes
+        -----
+        The `~yaml.UnsafeLoader` is used when parsing the YAML file.
+        """
+        try:
+            with open(path, "rb") as fd:
+                data = self._fromBytes(fd.read(), pytype)
+        except FileNotFoundError:
+            data = None
+
+        return data
+
+    def _fromBytes(self, serializedDataset: bytes, pytype: Optional[Type[Any]] = None) -> Any:
+        """Read the bytes object as a python object.
+
+        Parameters
+        ----------
+        serializedDataset : `bytes`
+            Bytes object to unserialize.
+        pytype : `class`, optional
+            Expected python type to be returned.
+
+        Returns
+        -------
+        inMemoryDataset : `object`
+            The requested data as an object, or None if the string could
+            not be read.
+        """
+        data = yaml.load(serializedDataset, Loader=yaml.FullLoader)
+
+        # This will be a simple dict so we need to convert it to
+        # the Filter type -- just needs the name
+        filter = pytype(data["canonicalName"])
+
+        return filter
+
+    def _writeFile(self, inMemoryDataset: Any) -> None:
+        """Write the in memory dataset to file on disk.
+
+        Will look for `_asdict()` method to aid YAML serialization, following
+        the approach of the simplejson module.
+
+        Parameters
+        ----------
+        inMemoryDataset : `object`
+            Object to serialize.
+
+        Raises
+        ------
+        Exception
```

Comment:
Please be more specific. We shouldn't be encouraging people to catch `Exception`.

Reply:
In the general sense of any formatter, we have no real idea what exceptions could be raised. It doesn't really matter either because all formatter exceptions are caught by datastore.